frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/index.tsx

```diff
@@ -66,14 +75,41 @@ function GeneButton({
           active={genesToDelete.includes(geneName)}
           font={currentFont}
         >
+          <InfoButtonWrapper
+            className={EXCLUDE_IN_SCREENSHOT_CLASS_NAME}
+            onClick={() => {
+              generateGeneInfo(geneName);
+
+              track(EVENTS.WMG_GENE_INFO, {
+                gene: geneName,
+              });
+
+              if (!dispatch) return;
+
+              // Clear cell type info here so that FMG panel closes
+              dispatch(clearCellInfoCellType());
```

Comment:
note that we clear cell info types here because when the gene info button is clicked on the xaxis then the FMG panel should also be cleared and replaced with the gene info panel

Reply:
discussion:

I wonder if we could centralize store mutation in reducer instead of burdening event handler to know the store mutation details, effectively following the best practice to use Redux pattern as an event store instead of a command bus

Concretely:

1. Crate an action "selectGenInfoFromXAxis"
2. Reducer gets `selectGenInfoFromXAxis` and add `geneInfo` and remove `cellInfoCellType`

This way the event handler only needs to dispatch `selectGenInfoFromXAxis` and doesn't need to care about what store mutation has to be done!

source:
https://engineering.tableau.com/redux-command-bus-or-event-store-2c4c044cd481#ab5e